src/main/java/pl/coderstrust/accounting/controller/InvoiceController.java

```diff
@@ -153,4 +161,21 @@ public void removeInvoice(@RequestBody int id) {
     invoiceService.updateInvoice(invoice);
     return ResponseEntity.ok().build();
   }
-}
\ No newline at end of file
+
+  @RequestMapping(value = "pdf/{id}", method = RequestMethod.GET, produces = MediaType.APPLICATION_PDF_VALUE)
+  @ApiOperation(value = "Returns invoice in pdf format")
+  public ResponseEntity<byte[]> invoicePdf(@PathVariable("id") int id) {
+    Optional<Invoice> invoiceFromDatabase = invoiceService.findById(id);
+    if (!invoiceFromDatabase.isPresent()) {
+      return ResponseEntity.notFound().build();
+    }
+    try {
+      ByteArrayOutputStream arrayOutputStream = new PdfService().generatePdf(PdfSet.createContent(
```

Comment:
You should not create new instance of PdfService - please use service injected by spring

Reply:
I would suggest to split that line into 2 lines - one creating content, other generating Pdf - now it's too long.

Reply:
done